### PR TITLE
fix: [CDS-74355]: Env and Infra Propagation fixes for delete/reorder and env inputs

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

Issues Fixed -->

1. Single infrastructure shows option to drag and re order.
2. For Env propagation, if the parent stage is deleted, UI breaks for stage propagating from it -- Video 2 & 3 below
3. Env Inputs don't appear properly for a stage which has env propagation - Video 1 below

#### Screenshots

Inputs -->

https://github.com/harness/harness-core-ui/assets/96409598/729cd8da-43ef-40b8-a261-deacf744552b

**Delete Stage ->**

https://github.com/harness/harness-core-ui/assets/96409598/69ceab02-8bb9-422d-8408-572d57e26cda

**Reorder stage -->** 

https://github.com/harness/harness-core-ui/assets/96409598/da207ab3-6368-445a-b50f-e41a6390c29e

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts [hunk 1]
```diff
@@ -737,7 +737,10 @@ export const getPropagatingStagesFromStage = (
       // IS NOT required as the useFromStage combinations are exclusive
       (currentStage.stage?.spec as DeploymentStageConfig)?.service?.useFromStage?.stage === stageId ||
       (currentStage.stage?.template?.templateInputs?.spec as DeploymentStageConfig)?.service?.useFromStage?.stage ===
-        stageId
+        stageId ||
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts:740]: We can refactor this properly and declare variables for repeated values. This is very hard to read and understand

File: src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts [hunk 2]
```diff
@@ -775,7 +786,10 @@ export const getParentPropagatedStage = (stage?: StageElementConfig): string | u
     (stage?.template?.templateInputs?.spec as DeploymentStageConfig)?.service?.useFromStage?.stage ||
     // The below mentioned conditions are for handling multi service propagation - Template/Non template Scenarios
     (stage?.spec as DeploymentStageConfig)?.services?.useFromStage?.stage ||
-    (stage?.template?.templateInputs?.spec as DeploymentStageConfig)?.services?.useFromStage?.stage
+    (stage?.template?.templateInputs?.spec as DeploymentStageConfig)?.services?.useFromStage?.stage ||
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts:789]: Same as above

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/utils/singleEnvironment/processSingleEnvironmentInitialValues.ts [hunk 3]
```diff
@@ -6,35 +6,71 @@
...
+        // In order to show environment and override inputs properly in a readonly manner, resolved values are required in formik
+        // for stage which propagates env data from a previous stage
+        const propagatedFromStage = (stages as StageElementWrapperConfig[])?.find(
+          stage => stage?.stage?.identifier === useFromStageValue
+        )
+        const environmentDetails = get(propagatedFromStage, 'stage.spec.environment')
+        if (!isEmpty(environmentDetails)) {
+          set(formState, 'environment', environmentDetails.environmentRef)
+
+          set(
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/utils/singleEnvironment/processSingleEnvironmentInitialValues.ts:47]: We can make this more readable by declaring values and then setting and use if else statement if nested ternary operator is used.

Comment (samarth-sharma31) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/utils/singleEnvironment/processSingleEnvironmentInitialValues.ts]: Done

File: src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts (hunk 2 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts]: Done

File: src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts]: Done

**Review by pankajsingh-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**